### ISSUE TITLE
(Feature Request)Deep Integration between Winget, Microsoft Store and Windows Update

### ISSUE THREAD

**Issue by dtantono** [microsoft/winget-cli#711]
I know winget update command is live and it is still experimental and also integration with Microsoft Store is still ongoing. So, is it possible in the future if we can update Winget installed apps, Microsoft Store apps and launch Windows Update simultaneously in a single command?

**Comment by denelon** [comment]
We are still looking at the best way to handle automatic updates with the package manager. I'll keep this feature to consider this kind of integration between the package manager and the store.

**Comment by Karl-WE** [comment]
I think that Store is a good idea with App Installer, although there are some issue with the Store not related to App Installer 
Example: Store app does not update apps in the background on many systems, especially those that aren't online for a long period of time like laptops, low bandwith.
Apps are listed as "pending" in the Store but do not download.

I regularly see the issue that store based apps are outdated. Unless the user stats up Store and click get updates, then all updates pending are processed. I have flagged this with the Windows Insider team and Store team but no success. This issue is going around since 1803 at least.

tl:dr Updating via Store is not reliable at the moment for users that do not use their devices often or have low bandwidth connections.

Preferring Windows Update would also enable the use on Windows Server see #702, but this would need to care to check and upgrade prerequisites to Winget.

**Comment by vedantmgoyal2009** [comment]
* @dtantono MS Store source is now available in Winget. You can post the Windows Update Feature Request under https://github.com/microsoft/winget-cli/issues/1863.

cc @denelon

**Comment by denelon** [comment]
@vedantmgoyal2009 this feature was created earlier than the other one. I'm going to keep this one open to look at the possibility of enabling Windows Update. If we get the version for the "msstore" source (not just the Win32 packages), `winget upgrade --all` would trigger updates for Microsoft Store packages.

**Comment by rokejulianlockhart** [comment]
@denelon, is what you describe—the absence of versions from the `msstore` source—tracked anywhere? It's a really annoying problem for me too for exactly the reason that you state.

**Comment by denelon** [comment]
Yes, and the work is in progress for getting version for packages in the "msstore" source. If you run `winget show` for one of the new Win32 Apps in the store you will see the version present. The Microsoft Store team is working on the remaining plumbing. This is a very large body of engineering work, and I don't have an ETA.

**Comment by IngwiePhoenix** [comment]
Came here while googling for integrating Winget into Windows Update.

Any progress on that part in particular? Thanks!

**Comment by denelon** [comment]
@IngwiePhoenix,

Not yet. That's a fairly large undertaking with several other compliance concerns. We would likely split that out into its own issue before tackling it in earnest.

**Comment by rokejulianlockhart** [comment]
@denelon and @IngwiePhoenix, if you happen to create a separate issue, please link it here.

**Comment by jdwalker-emis** [comment]
@rokejulianlockhart  Added #3188 to split off the windows update integration.

**Comment by denelon** [comment]
Thanks for the suggestion! Sharing where things stand today and how we're thinking about the remaining gaps.

**What works today**

WinGet can already update packages across every configured source in a single command:

```
winget upgrade --all
```

This includes packages from the Microsoft Store (`msstore` source) alongside packages from the community repository and any other configured sources.

**Known gaps we're still working through**

1. **Store packages without version metadata.** Some packaged applications delivered through the Microsoft Store don't surface version information to WinGet, so we can't reliably detect that an update is available. Closing this gap depends on richer metadata being available for those packages.

2. **A way to explicitly refresh *all* Store apps via WinGet** — including those that aren't source-mapped to a WinGet package. This is an open request we're evaluating; today WinGet only acts on packages it can correlate to a known source entry.

3. **Windows Update integration (both directions).**
   - *Windows Update -> WinGet:* We're working with the Update Orchestrator so Windows Update can include a check for WinGet-managed package updates. This is in progress.
   - *WinGet -> Windows Update:* Having WinGet explicitly invoke/trigger Windows Update is still an open request we're evaluating.

We'll keep this issue open to track the remaining evaluation items (explicit "update all Store apps" behavior and WinGet-initiated Windows Update). Appreciate the feedback!

_(Drafted with GitHub Copilot assistance.)_
